packages/waku/src/lib/rsc/worker-impl.ts

```diff
@@ -87,30 +91,47 @@ const handleRender = async (mesg: MessageReq & { type: 'render' }) => {
 };
 
 const dummyServer = new Server(); // FIXME we hope to avoid this hack
-const vitePromise = viteCreateServer({
-  plugins: [
-    nonjsResolvePlugin(),
-    rscTransformPlugin(false),
-    rscReloadPlugin((type) => {
-      const mesg: MessageRes = { type };
-      parentPort!.postMessage(mesg);
-    }),
-    rscDelegatePlugin((source) => {
-      const mesg: MessageRes = { type: 'hot-import', source };
-      parentPort!.postMessage(mesg);
-    }),
-  ],
-  ssr: {
-    resolve: {
-      conditions: ['react-server', 'workerd'],
-      externalConditions: ['react-server', 'workerd'],
+
+const resolvedViteConfig = await resolveViteConfig({}, 'serve');
+const mergedViteConfig = await mergeViteConfig(
+  {
+    ...resolvedViteConfig,
+
+    plugins: resolvedViteConfig.plugins.filter(
+      (plugin) => !plugin.name.startsWith('vite:'),
```

Comment:
Again, I wonder why we need this and if we need both in two places.

Reply:
It's because we don't vite default plugins to be mentioned two times. For instance react refresh would fail because there would two instances of it.

Reply:
https://github.com/dai-shi/waku/pull/240#discussion_r1424492002

> React refresh would fail

I see. I will play with it later. Does it mean there's only one vite server that keep vite plugins? (Can't guess which from the diff, and I don't remember how many they are...)

Please extract this patch/merge logic in a separate file and reuse it. something like `packages/waku/src/lib/plugins/patch-react-refresh.ts`.

Reply:
I don't exactly mean patching plugin. It should be a reusable logic and preferably add some comments for the intention.